src/OpenTelemetry.Contrib.Instrumentation.AWS/Implementation/AWSMessagingUtils.cs

```diff
@@ -14,22 +14,23 @@
 // limitations under the License.
 // </copyright>
 
+using System;
 using System.Collections.Generic;
 using OpenTelemetry.Context.Propagation;
 
 namespace OpenTelemetry.Contrib.Instrumentation.AWS.Implementation;
 
 internal static class AWSMessagingUtils
 {
-    internal static void Inject(IRequestContextAdapter request, PropagationContext propagationContext)
+    internal static void Inject(Action<IReadOnlyDictionary<string, string>>? addAttributes, PropagationContext propagationContext)
     {
-        if (!request.CanInject)
+        if (addAttributes == null)
```

Comment:
Is this check really necessary? I don't think `null` is possibly passed here, as one can easily verify. If null is actually passed, it will be accidental and this check will make the bug harder to find.